subsys/pm/device_runtime.c

```diff
@@ -119,6 +119,10 @@ int pm_device_runtime_get(const struct device *dev)
 	int ret = 0;
 	struct pm_device *pm = dev->pm;
 
+	if (pm == NULL) {
+		return -ENOSYS;
+	}
+
```

Comment:
Checking for this is something good imho. Just a few points with the implementation:

1) It seems to me that it should return `-ENOTSUP` since runtime API was enabled and is a specific device that does not support PM.
2) Why don't we use `CHECKIF` here and make this check optional.

Other than these points I think the commit is nice.

Reply:
agreed, should be `-ENOTSUP`.

Reply:
Done. Not using `CHECKIF` because it should never be promoted to an assertion.
Some device drivers just don't need to support PM because they are always in their lowest power state when not running.

An example use case is a common thread that can be given any temperature sensor to run. Giving it a sensor that doesn't have PM is not an error condition, but it still needs to run `pm_device_runtime` functions for sensors that do.

Reply:
That is ok, I still think `CHECKIF` is better. The default behavior is the branch, it is really expected that if someone promotes it to an assert or to a NOP they know what they are doing. But that is minor.

Reply:
I disagree, doing a spot check of other uses in-tree, they are all for conditions which either should not occur at runtime (e.g. passing a NULL device as an argument) or are the result of an invalid configuration (e.g. some name is too long).

These are the results of bugs or misconfigurations, and you could reasonably select `CONFIG_ASSERT_ON_ERRORS` after initial testing and expect nothing to go wrong. A device not supporting PM is not a bug or a misconfiguration, and in the case of upstream drivers can't be fixed without copying the driver.

You would be unable to use `CONFIG_ASSERT_ON_ERRORS` or `CONFIG_NO_RUNTIME_CHECKS` if any particular device did not support PM.